example/spec/loadTest/runScaledTest.js

```diff
@@ -0,0 +1,362 @@
+const { LambdaStep } = require('@cumulus/integration-tests/sfnStep');
+const { listS3ObjectsV2 } = require('@cumulus/aws-client/S3');
+const { providers, collections, granules } = require('@cumulus/api-client');
+const { randomString } = require('@cumulus/common/test-utils');
+
+const {
+  buildAndExecuteWorkflow,
+  api: apiTestUtils,
+} = require('@cumulus/integration-tests');
+
+const {
+  createTestSuffix,
+  createTimestampedTestId,
+  loadConfig,
+} = require('../helpers/testUtils');
+
+// ** Configurable Variables
+const expectedGranuleCount = 500; // Per batch
+const batches = 5; // Number of batches to run
+const providerPathTemplate = `ingest_${expectedGranuleCount}_test`; // Directory pattern to use
+const providerPaths = generateIterableTestDirectories(providerPathTemplate, batches); // Given N batches, run over "pattern_0-9,a-z" directories
+// const providerPaths = ['ingest_100_test_0'];
+
+const waitForIngestTimeoutMs = 6 * 60 * 1000;
+const statsTimeout = 240 * 1000;
+const granuleCountThreshold = 0.95;
+
+const expectedMaxWorkflowInvocation = expectedGranuleCount * providerPaths.length; // Number of Ingest workflow invocations.
+const publishGranulesMinInvocations = (expectedGranuleCount * providerPaths.length) / 10; // Number of publish invocations to require to pass metrics.
+const publishExecutionsMinInvocations = (expectedGranuleCount * providerPaths.length) / 10; // Number of execution publish invocations to require to pass metrics
+
+const rdsClusterName = process.env.rdsClusterName || 'cumulus-dev-rds-cluster';
+const testCollections = [];
+let allIngestedGranules = [];
+let beforeAllCompleted;
+let bucket;
+let testConfig;
+let provider;
+let stackName;
+let workflowExecution;
+
+const testBeginTime = new Date(Date.now() - 60000);
+jasmine.DEFAULT_TIMEOUT_INTERVAL = process.env.LOAD_TEST_TIMEOUT || 4200000;
+
+
+const checkGranuleCount = async (granuleCollection, config, count) => {
+  console.log(`Using collection ${granuleCollection.name}___${granuleCollection.version}`);
+  let prevGranuleFilesCount;
+  let granuleFiles = [];
+  try {
+    await pWaitFor(async () => {
+      prevGranuleFilesCount = granuleFiles.length;
+      granuleFiles = await listS3ObjectsV2({
+        Prefix: `${granuleCollection.name}___${granuleCollection.version}`,
+        Bucket: config.buckets.protected.name,
+      });
+      console.log(`Ingested granules found: ${granuleFiles.length}`);
+      if (granuleFiles.length > count) {
+        console.log(`Test misconfiguration detected - ${granuleFiles.length} granules are present when there should be max ${count}`);
+        return true;
+      }
+      return granuleFiles.length === count && prevGranuleFilesCount === count;
+    },
+    { interval: 20000, timeout: waitForIngestTimeoutMs });
+  } catch (error) {
+    if (error.name !== 'TimeoutError') {
+      throw error;
+    }
+    console.log(`Full Granules not ingested: ${prevGranuleFilesCount}`);
```

Comment:
Is there no other possible error that could occur or are we swallowing other ones on pupose for the sake of the test?

Reply:
The conditional is if *not* TimeoutError, throw. 

We're explicitly swallowing a TimeoutError because that's what pWaitFor throws if the timeout occurs.   If it times out, we have it return the count and let the caller decide what to do about the count total.